### ISSUE TITLE
Black windows; --replace gives a segfault

### ISSUE THREAD

**Issue by gauravjuvekar** [linuxmint/Cinnamon#3388]
LMDE UP8 64-bit
Cinnamon 2.0.14

Not exactly reproducible but all windows suddenly turn black apparently randomly. They are still resizable and the cinnamon ui (panels/menu) works correctly. The window borders(including titlebar) also turn black. On issuing `r` in Alt+F2 prompt, the windows appear normally but without borders/titlebars and are not resizable. Cinnamon segfaults at this point(cinnamon ui elements not visible).

Here's what happens when issuing `cinnamon --replace` in a terminal. Logging out and back in resolves all issues.

```
$ cinnamon --replace
      JS LOG: About to start Cinnamon
      JS LOG: Cinnamon started at Wed Jul 23 2014 20:20:37 GMT+0530 (IST)
      JS LOG: network applet: Cannot find connection for active (or connection cannot be read)

(cinnamon:6190): St-CRITICAL **: _st_create_shadow_material: assertion `src_texture != COGL_INVALID_HANDLE' failed

(cinnamon:6190): St-CRITICAL **: _st_create_shadow_material: assertion `src_texture != COGL_INVALID_HANDLE' failed

(cinnamon:6190): St-CRITICAL **: _st_create_shadow_material: assertion `src_texture != COGL_INVALID_HANDLE' failed
Segmentation fault
$ 
```

**Comment by Vahan86** [comment]
@gauravjuvekar, does this issue still occur in Cinnamon 3.0.x?

**Comment by gauravjuvekar** [comment]
Nope, seems to be fixed in 3.0.7 on mint 18.
